opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/InterfaceToNodeCacheDaoImpl.java

```diff
@@ -176,9 +182,40 @@ public int compareTo(final Value that) {
     @Autowired
     private IpInterfaceDao m_ipInterfaceDao;
 
+    @Autowired
+    private TransactionOperations transactionOperations;
+
     private final ReadWriteLock m_lock = new ReentrantReadWriteLock();
     private final SortedSetMultimap<Key, Value> m_managedAddresses = Multimaps.newSortedSetMultimap(Maps.newHashMap(), TreeSet::new);
 
+    private final Timer refreshTimer = new Timer(getClass().getSimpleName());
+
+    // in ms
+    private final long refreshRate;
+
+    public InterfaceToNodeCacheDaoImpl() {
+        this(-1); // By default refreshing the cache is disabled
+    }
+
+    public InterfaceToNodeCacheDaoImpl(long refreshRate) {
+        this.refreshRate = refreshRate;
+    }
+
+    @PostConstruct
+    public void init() {
+        // we call this, to ensure the bean-initialization is blocked, until it is initialized
+        dataSourceSync();
+
+        if (refreshRate > 0) {
+            refreshTimer.schedule(new TimerTask() {
+                @Override
+                public void run() {
+                    dataSourceSync();
```

Comment:
Should we also catch and log run-time exceptions here so that the timer continues to fire even if there is a temporary blip?

Reply:
That is a good point, I guess that makes sense.